tests/Aspire.Hosting.Containers.Tests/FromDockerfileTests.cs

```diff
@@ -0,0 +1,449 @@
+        Assert.Equal(tempContextPath, annotation.ContextPath);
+        Assert.Equal(tempDockerfilePath, annotation.DockerfilePath);
+    }
+
+    [Fact]
+    public async Task WithDockerfileWithValidContextPathValidDockerfileWithExplicitDefaultNameSucceeds()
+    {
+        using var builder = TestDistributedApplicationBuilder.Create();
+        var (tempContextPath, tempDockerfilePath) = await CreateTemporaryDockerfileAsync();
+
+        var container = builder.AddContainer("mycontainer", "myimage")
+                               .WithDockerfile(tempContextPath, "Dockerfile");
+
+        var annotation = Assert.Single(container.Resource.Annotations.OfType<DockerfileBuildAnnotation>());
+        Assert.Equal(tempContextPath, annotation.ContextPath);
+        Assert.Equal(tempDockerfilePath, annotation.DockerfilePath);
+    }
+
+    [Fact]
+    public async Task WithDockerfileWithValidContextPathValidDockerfileWithExplicitNonDefaultNameSucceeds()
+    {
+        using var builder = TestDistributedApplicationBuilder.Create();
+        var (tempContextPath, tempDockerfilePath) = await CreateTemporaryDockerfileAsync("Otherdockerfile");
+
+        var container = builder.AddContainer("mycontainer", "myimage")
+                               .WithDockerfile(tempContextPath, "Otherdockerfile");
+
+        var annotation = Assert.Single(container.Resource.Annotations.OfType<DockerfileBuildAnnotation>());
+        Assert.Equal(tempContextPath, annotation.ContextPath);
+        Assert.Equal(tempDockerfilePath, annotation.DockerfilePath);
+    }
+
+    [Fact]
+    public async Task WithDockerfileWithValidContextPathValidDockerfileWithExplicitAbsoluteDefaultNameSucceeds()
+    {
+        using var builder = TestDistributedApplicationBuilder.Create();
+        var (tempContextPath, tempDockerfilePath) = await CreateTemporaryDockerfileAsync();
+
+        var container = builder.AddContainer("mycontainer", "myimage")
+                               .WithDockerfile(tempContextPath, tempDockerfilePath);
+
+        var annotation = Assert.Single(container.Resource.Annotations.OfType<DockerfileBuildAnnotation>());
+        Assert.Equal(tempContextPath, annotation.ContextPath);
+        Assert.Equal(tempDockerfilePath, annotation.DockerfilePath);
+    }
+
+    private static async Task<(string ContextPath, string DockerfilePath)> CreateTemporaryDockerfileAsync(string dockerfileName = "Dockerfile", bool createDockerfile = true)
+    {
+        var tempContextPath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+        Directory.CreateDirectory(tempContextPath);
+
+        var tempDockerfilePath = Path.Combine(tempContextPath, dockerfileName);
+
+        if (createDockerfile)
+        {
+            await File.WriteAllTextAsync(tempDockerfilePath, HelloWorldDockerfile);
+        }
+
+        return (tempContextPath, tempDockerfilePath);
+    }
+
+    private static bool IsDockerAvailable()
```

Comment:
Don't we have something better for this? .NET must have something for this.

cc @joperezr @eerhardt

Reply:
We have a `[RequiresDocker]` attribute which can be applied to tests. Looking at both places where this is used, I think it would be better to just use that as opposed to this. The implementation of that attribute is far simpler as it isn't actually checking if docker itself is running, but instead just makes the assumption based on the fact that we are running on Helix. https://github.com/dotnet/aspire/blob/b891c21433a6acf07ead8725fbd99825d7a21ea6/tests/Aspire.Components.Common.Tests/RequiresDockerAttribute.cs#L12

Reply:
@mitchdenny was trying this but ran into some issues. I will take a look at this to check what might be missing.

Reply:
So, I've tried `[RequiresDocker]`. It didn't work. Here are my requirements:

1. Must allow the test to run locally.
2. Must allow the test to run on the build agent on Linux
3. Must allow the test to run on the helix machines on Linux
4. Must not require a PhD in MSBuild to make it work (we want a pattern that others can follow for other resource types).

I don't want to bog this feature down in making this work because this is a long standing issue that is blocking a lot of other tests running in CI as well. But I didn't want to merge this feature without at least some coverage which is why I made it work this way.

Reply:
I'm trying out a fix. The `IsSupported` condition did not expect the tests to be running on the build machine. It came from `RequiresDockerTheoryAttribute` which was only for helix, but then we had merged the attributes. I'll push a fix for that.

Reply:
@radical I accidentally blew away your change after a force push (I was merging in some changes from main and accidentally assumed that I had a conflicting change that I might have already main upstream).

Feel free to force push over my change because your updates would have included what I was doing anyway.

Reply:
OK I've put your changes back in. State of git-fu zen achieved.